### PR TITLE
fix(components/lookup): add required label asterisk for template driven forms

### PR DESCRIPTION
**Pull request by Blackbaud-ErikaMcVey**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2694/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2694/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2694/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2694/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2694/playground/)

**Review by Blackbaud-PaulCrowder — DISMISSED** [review]

File: libs/components/forms/src/lib/modules/input-box/input-box.component.ts [hunk 1]
```diff
@@ -248,6 +252,11 @@ export class SkyInputBoxComponent
 
   public ngOnInit(): void {
     this.#inputBoxHostSvc.init(this);
+
+    this.#inputBoxHostSvc.required.subscribe((required) => {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/input-box/input-box.component.ts]: Need a `takeUntil`?

File: libs/components/forms/src/lib/modules/input-box/input-box-host.service.ts [hunk 2]
```diff
@@ -11,6 +11,9 @@ import { SkyInputBoxComponent } from './input-box.component';
 @Injectable()
 export class SkyInputBoxHostService {
   #host: SkyInputBoxComponent | undefined;
+  #requiredSubject = new BehaviorSubject<boolean>(false);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/input-box/input-box-host.service.ts:14]: Needs to be unsubscribed in `ngOnDestroy`.

File: libs/components/forms/src/lib/modules/input-box/input-box-host.service.ts [hunk 3]
```diff
@@ -74,4 +77,8 @@ export class SkyInputBoxHostService {
 
     this.#host.setHintTextScreenReaderOnly(hide);
   }
+
+  public setRequired(required: boolean): void {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/input-box/input-box-host.service.ts:90]: Can you put a detailed JSDoc comment above this method explaining why another component would need it? I had to dig into the inner workings of input box to understand why using Angular's `required` directive wouldn't "just work" for the lookup component's specific case.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/2711

**Pull request merged**